### PR TITLE
datasets split_train_val() parameter error

### LINKED ISSUE #57: datasets split_train_val() parameter error
### System Info

Bender, Python environment created by `distributed/slurm/create_venv_bender.sh` 

### Who can help?

_No response_

### Information

- [x] An existing script in the repository (e.g. under `distributed/`, `data/`, `evals/`)
- [ ] My own modified or new scripts

### Area

- [ ] Pretraining (`distributed/`)
- [ ] Post-training / alignment (`alignment/`)
- [x] Data processing / tokenization (`data/`): `data/filters/train_annotator.py`
- [ ] Evaluation (`evals/`)
- [ ] Synthetic data (`synthetic/`)
- [ ] Model merging (`merge/`)
- [ ] Tokenizer (`tokenizer/`)
- [ ] Other / tools (`tools/`)

### Reproduction

1. Create dataset `data.jsonl` of size `n`, such that `n mod 10 != 0`.
2. Run `data/filters/train_annotator.py` with parameter `--train_dataset_dir data.jsonl`.

#### Encountered Error

We encounter a parameter type error while creating the split into train and test set. We set the size of the test set as `n * 0.1` which leads to a non-integer number. However, the parameter `test_size` in the function `datasets.Dataset.train_test_split()` only accepts either type `float` between `0.0` and `1.0` or type `int` . See [docs](https://huggingface.co/docs/datasets/main/en/package_reference/main_classes#datasets.Dataset.train_test_split).
```bash
  # Split the dataset into train and test sets.
  dataset = dataset.train_test_split(
      test_size=min(
          args.test_size, len(dataset) * 0.1       # <- this can be non-integer > 1.0
      ),  # Ensure test_size doesn't exceed dataset size
      seed=args.seed,
      stratify_by_column=args.target_column,
  )
```

#### Proposed Fix
Cast `len(dataset) * 0.1` to `int`:
```bash
  dataset = dataset.train_test_split(
      test_size=min(
          args.test_size, int(len(dataset) * 0.1)
      ),
     ...,
  )
```

### Expected behavior

We expect the training to run uninterrupted.

### PR DESCRIPTION
**Pull request by taoshende**
# What does this PR do?

Fixes #57 

Casts parameter to `int` before calling `datasets.Dataset.train_test_split()` to prevent parameter type error.

## Before submitting
- [x] I have read [CONTRIBUTING.md](https://github.com/Polygl0t/llm-foundry/blob/main/CONTRIBUTING.md).
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] The change was discussed/approved via a GitHub issue (please add a link if so).
- [x] The title and description clearly explain what was changed and why.
- [x] Commits have been squashed into a single, clean commit.
- [x] Existing tests still pass (`pytest` from the repository root).
- [ ] New functionality is covered by tests (if applicable).
- [x] I have run `pre-commit run --all-files` and fixed any issues.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to @-mention
the maintainers (@Nkluge-correa or @shiza-fatimah) or other contributors who may be interested in
your PR.

### CODE REVIEW

**Comment by Nkluge-correa** [comment]
Hi @taoshende, thanks for the fix. This looks correct and ready to merge. --test_size is an int, and len(dataset) * 0.1 is a float, so min() could previously return a float (e.g., 250.0) that datasets.train_test_split() rejects. Casting the cap to int resolves the type error without changing the common path where args.test_size is the smaller value.

Merged. 👍
